packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -418,14 +439,30 @@ class Slider extends React.Component {
       [classes.vertical]: vertical,
     });
 
-    const thumbClasses = classNames(classes.thumb, commonClasses);
-
     const trackProperty = vertical ? 'height' : 'width';
     const thumbProperty = vertical ? 'top' : 'left';
     const inlineTrackBeforeStyles = { [trackProperty]: this.calculateTrackBeforeStyles(percent) };
     const inlineTrackAfterStyles = { [trackProperty]: this.calculateTrackAfterStyles(percent) };
     const inlineThumbStyles = { [thumbProperty]: `${percent}%` };
 
+    /** Start Thumb Icon Logic Here */
+    const withIcon = !!thumbIcon;
+    const ThumbIcon = withIcon
+      ? React.cloneElement(thumbIcon, {
+          ...thumbIcon.props,
+          className: `${classes.thumbIcon} ${thumbIcon.props.className || ''}`,
```

Comment:
`classNames(thumbIcon.props.className, classes.thumbIcon)` is equivalent here. We should stick with `classNames`.

Reply:
![demo-slider-icon](https://user-images.githubusercontent.com/16651811/44473687-50e86b80-a64a-11e8-8ab8-f8d34c256bd3.gif)

Previously there was an error at my part, I messed some CSS up for this. Which was causing the active, jumped states to not show properly. Which I fixed in this iteration.

Regarding this change, I am doing the following when no thumb prop is provided.

```
classes.thumb: {
   width: 12,
   height: 12,
   backgroundColor: colors.primary,
}
```

But when I provide it with the thumb prop, I give it some extra height & width. Because then the image/svg icon are no longer visible

```
classes.thumbIconWrapper: {
    width: 17,
    height: 17,
    backgroundColor: 'transparent',
}
```

Is the above approach correct, or should they be of default size which is 12px (height & width) whether a custom thumb prop is provided or not?

> classNames(thumbIcon.props.className, classes.thumbIcon) is equivalent here. We should stick with classNames.

Fixed this change. Thank you for the suggestion.

Reply:
> I guess we do the ts definitions in another PR or is @adeelibr ok with adding a line in Slider.d.ts?

I am not good with .ts, but I would love to jump in. I'll try to do in another PR once this is done.

> To get CI working merge master to get #12611

Once the changes have been finalized, I'll merge this PR with master in the next commit.

Reply:
> Should I change this to thumbIcon too, along with inside the code as well.
> Makes sense to me. @eps1lon?

@eps1lon would it be better if I changed the `thumb` prop to `thumbIcon`, @mbrookes thinks it is better. Your opinion.

Reply:
Follow up! 
I merged masted into the branch.

Reply:
As a comparison. This is the custom override with height/width 17: and on active/jump state height/width is 22
![slider-custom](https://user-images.githubusercontent.com/16651811/44604264-bda46700-a7fe-11e8-8767-2f79f09650c8.gif)

Here is the default settings with height/width 12; and on active/jump state etc height/width is 17
![slider-default](https://user-images.githubusercontent.com/16651811/44604300-d7de4500-a7fe-11e8-9418-0df351231a52.gif)

Currently this PR has a custom height/width of 17; i.e, the first image.
This is the only confusion I have left, any feedback would really put me in the right direction. Thank you.